src/isar/state_machine/state_machine.py

```diff
@@ -328,17 +334,101 @@ def _mission_started(self) -> None:
+        if self.run_mission_by_task:
+            self.current_task.status = TaskStatus.InProgress
+        self.current_mission.status = MissionStatus.InProgress
+        self.publish_task_status(task=self.current_task)
+        self.logger.info(
+            f"Successfully initiated "
+            f"{type(self.current_task).__name__} "
+            f"task: {str(self.current_task.id)[:8]}"
+        )
+
+    def _try_initiate_task_or_mission(self) -> bool:
+        retries = 0
+        started_mission = False
+        try:
+            while not started_mission:
+                try:
+                    if self.run_mission_by_task:
+                        self.robot.initiate_task(self.current_task)
+                    else:
+                        self.robot.initiate_mission(self.current_mission)
+                except RobotException as e:
+                    retries += 1
+                    self.logger.warning(
+                        f"Initiating failed #: {str(retries)} "
+                        f"because: {e.error_description}"
+                    )
+
+                    if retries >= settings.INITIATE_FAILURE_COUNTER_LIMIT:
+                        self.current_task.error_message = ErrorMessage(
+                            error_reason=e.error_reason,
+                            error_description=e.error_description,
+                        )
+                        self.logger.error(
+                            f"Mission will be cancelled after failing to initiate "
+                            f"{self.initiate_failure_counter_limit} times because: "
+                            f"{e.error_description}"
+                        )
+                        self._initiate_failed()
+                        return False
+                started_mission = True
+        except RobotInfeasibleTaskException as e:
+            self.current_task.error_message = ErrorMessage(
+                error_reason=e.error_reason, error_description=e.error_description
+            )
+            self.logger.warning(
+                f"Failed to initiate task "
+                f"{str(self.current_task.id)[:8]} after retrying "
+                f"{retries} times because: "
+                f"{e.error_description}"
+            )
+            self._initiate_infeasible()
+            # We only fail the transition back to monitor if we are unable to continue the mission
+            return True
+
+        except RobotInfeasibleMissionException as e:
+            self.current_mission.error_message = ErrorMessage(
+                error_reason=e.error_reason, error_description=e.error_description
+            )
+            self.logger.warning(
+                f"Failed to initiate mission "
+                f"{str(self.current_mission.id)[:8]} because: "
+                f"{e.error_description}"
+            )
+            self._initiate_failed()
+            return False
+        return True
+
+    def _try_start_mission(self) -> bool:
```

Comment:
This contains the functionality that used to the in the initiate and intialisation states